File: Módulo 1 - C/Mario (Hard)/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int size;

    do
    {
        size = get_int("Enter tower size: ");
    }
    while(size < 1 || size > 8);

    for(int i = 0; i < size; i++)
    {
        int spacesCount = size - (i + 1);
        int builtWalls = 0;

        for(int x = 0; x < size; x++)
        {
            if(spacesCount > 0)
            {
                printf(" ");
                spacesCount--;
            }
            else
            {
                printf("#");
                builtWalls++;
            }
        }

        printf("  ");

        for(int x = 0; x < builtWalls; x++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: Módulo 6 - Python/mario/less/mario.py
from cs50 import get_int

while True:
    size = get_int("Height: ")
    if 0 < size < 9:
        break

for i in range(size):
    air = size - i - 1

    for j in range(air):
        print(" ", end="")

    for k in range(i + 1):
        print("#", end="")

    print()